bundles/org.openhab.binding.haassohnpelletstove/src/main/java/org/openhab/binding/haassohnpelletstove/internal/HaasSohnpelletstoveHandler.java

```diff
@@ -0,0 +1,327 @@
+                }
+            } else {
+                logger.debug("Error. Command is the wrong type: {}", command.toString());
+            }
+        } else if (channelUID.getId().equals(CHANNELECOMODE)) {
+            String postData = null;
+            if (command.equals(OnOffType.ON)) {
+                postData = "{\"eco_mode\":true}";
+            } else if (command.equals(OnOffType.OFF)) {
+                postData = "{\"eco_mode\":false}";
+            }
+            if (postData != null) {
+                logger.debug("Executing {} command", CHANNELECOMODE);
+                updateOvenData(postData);
+            }
+        }
+    }
+
+    /**
+     * Calls the service to update the oven data
+     *
+     * @param postdata
+     */
+    private boolean updateOvenData(@Nullable String postdata) {
+        Helper message = new Helper();
+        if (serviceCommunication.updateOvenData(postdata, message, this.getThing().getUID().toString())) {
+            updateStatus(ThingStatus.ONLINE);
+            return true;
+        } else {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.OFFLINE.COMMUNICATION_ERROR,
+                    message.getStatusDesription());
+            return false;
+        }
+    }
+
+    @Override
+    public void initialize() {
+        logger.info("Initializing haassohnpelletstove handler for thing {}", getThing().getUID());
+        config = getConfigAs(HaasSohnpelletstoveConfiguration.class);
+        boolean validConfig = true;
+        String errors = "";
+        String statusDescr = null;
+        if (config.refreshRate < 0 && config.refreshRate > 999) {
+            errors += " Parameter 'refresh Rate' greater then 0 and less then 1000.";
+            statusDescr = "Parameter 'refresh Rate' greater then 0 and less then 1000.";
+            validConfig = false;
+        }
+        if (config.hostIP == null) {
+            errors += " Parameter 'hostIP' must be configured.";
+            statusDescr = "IP Address must be configured!";
+            validConfig = false;
+        }
+        if (config.hostPIN == null) {
+            errors += " Parameter 'hostPin' must be configured.";
+            statusDescr = "PIN must be configured!";
+            validConfig = false;
+        }
+        errors = errors.trim();
+        Helper message = new Helper();
+        message.setStatusDescription(statusDescr);
+        if (validConfig) {
+            serviceCommunication.setConfig(config);
+            if (serviceCommunication.refreshOvenConnection(message, this.getThing().getUID().toString())) {
+                if (updateOvenData(null)) {
+                    updateStatus(ThingStatus.ONLINE);
+                    updateLinkedChannels();
+                }
+            } else {
+                logger.info("Setting thing '{}' to OFFLINE: {}", getThing().getUID(), errors);
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, message.getStatusDesription());
```

Comment:
The log message could be removed, as the status update is already logged by the framework.

The state change originated by updateStatus() is logged to events.log. Including the status detail message.

Please check all.

Reply:
ok